File: tasks/implementations/node/tasks/third_party_api.js
const fs = require('fs');
module.exports = function(size, fixturesRoot) {
  const text = fs.readFileSync(`${fixturesRoot}/mocks/twitter_like_${size}.json`, 'utf8');
  let total = 0;
  for (const match of text.matchAll(/"id":\s*"([^"]+)",\s*"text":\s*"([^"]+)",\s*"likes":\s*(\d+)/g)) {
    total += match[1].length + match[2].length + Number(match[3]);
  }
  console.log(total);
};

File: tasks/implementations/go/tasks/third_party_api.go
package main

import (
    "fmt"
    "os"
    "path/filepath"
    "regexp"
    "strconv"
)

func RunThirdPartyApi(size string, fixturesRoot string) {
    data, _ := os.ReadFile(filepath.Join(fixturesRoot, "mocks", "twitter_like_"+size+".json"))
    re := regexp.MustCompile(`"id":\s*"([^"]+)",\s*"text":\s*"([^"]+)",\s*"likes":\s*(\d+)`)
    total := 0
    for _, match := range re.FindAllStringSubmatch(string(data), -1) {
        likes, _ := strconv.Atoi(match[3])
        total += len(match[1]) + len(match[2]) + likes
    }
    fmt.Println(total)
}
